pgjdbc/src/test/java/org/postgresql/test/jdbc2/ResultSetTest.java

```diff
@@ -716,4 +719,81 @@ public void testUpdateWithPGobject() throws SQLException {
     stmt.close();
   }
 
+  public void testResultSetColumnMappingCache() throws SQLException {
+    // Plain statement
+    Statement stmt = con.createStatement();
+    ResultSet rs = stmt.executeQuery("select * from testrs");
+    HashMap<String, Integer> columnNameIndexMap;
+    columnNameIndexMap = getResultSetColumnNameIndexMap(rs);
+    assertEquals(null, columnNameIndexMap);
+    assertTrue(rs.next());
+    columnNameIndexMap = getResultSetColumnNameIndexMap(rs);
+    assertEquals(null, columnNameIndexMap);
+    rs.getInt("ID");
+    columnNameIndexMap = getResultSetColumnNameIndexMap(rs);
+    assertNotNull(columnNameIndexMap);
+    rs.getInt("id");
+    assertSame(columnNameIndexMap, getResultSetColumnNameIndexMap(rs));
+    rs.close();
+    rs = stmt.executeQuery("select * from testrs");
+    columnNameIndexMap = getResultSetColumnNameIndexMap(rs);
+    assertEquals(null, columnNameIndexMap);
+    assertTrue(rs.next());
+    rs.getInt("Id");
+    columnNameIndexMap = getResultSetColumnNameIndexMap(rs);
+    assertNotNull(columnNameIndexMap);
+    rs.close();
+    stmt.close();
+    // Prepared statement
```

Comment:
"prepared statement" deserves its own test method.
It is typically better have "27 of 67 failures" than "1 of 1 failure".
With more fine granular tests it might be easier to track the progress & debug.

However, I do not ask to always split each assertion to its own test method.